pkg/prometheus/promcfg.go

```diff
@@ -3409,3 +3410,14 @@ func (cg *ConfigGenerator) getScrapeClassOrDefault(name *string) *monitoringv1.S
 	}
 	return nil
 }
+
+func getLowerByteSize(x, y monitoringv1.ByteSize) monitoringv1.ByteSize {
```

Comment:
to make it less error-prone, it would be better to pass the service/pod monitor value + CommonPrometheusFields

```suggestion
func getBodySizeLimit(v *monitoringv1.ByteSize, cpf CommonPrometheusFields) *monitoringv1.ByteSize {
```